cli/resources.rs

```diff
@@ -247,21 +190,47 @@ impl DenoAsyncRead for Resource {
   fn poll_read(&mut self, buf: &mut [u8]) -> Poll<usize, ErrBox> {
     let mut table = RESOURCE_TABLE.lock().unwrap();
     let repr = table.get_mut(&self.rid).ok_or_else(bad_resource)?;
```

Comment:
I'm still iterating on the API but I found this bit tricky. I'd like to be able to downcast `dyn DenoResource` to `DenoAsyncRead`, but `DenoAsyncRead` is a trait object it's not possible.
```
// what I'd like to do:
  resources::with_mut_resource(&self.rid, move |resource| {
      let resource = resource
        .downcast_mut::<dyn DenoAsyncRead>()
        .ok_or_else(bad_resource)?;
      f.0.poll_read(buf).map_err(ErrBox::from)
    })
```

@piscisaureus any advice?